tangelo/linq/simulator.py

```diff
@@ -26,515 +26,64 @@
 (e.g a number of shots is required).
 
 Some backends may only support a subset of the above. This information is
-contained in a separate data-structure.
+contained in the static method backend_info for each backend.
 """
 
-import os
-import math
-from collections import Counter
+from typing import Union, Tuple, Type
 
-import numpy as np
-from scipy import stats
-from bitarray import bitarray
-from openfermion.ops import QubitOperator
+import numpy
 
+from tangelo.linq.simulator_base import SimulatorBase
+from tangelo.linq.target import target_dict, backend_info
+from tangelo.linq.noisy_simulation import NoiseModel
 from tangelo.helpers.utils import default_simulator
-from tangelo.linq import Gate, Circuit
-from tangelo.linq.helpers.circuits.measurement_basis import measurement_basis_gates
-import tangelo.linq.translator as translator
 
 
-# Data-structure showing what functionalities are supported by the backend, in this package
-backend_info = dict()
-backend_info["qiskit"] = {"statevector_available": True, "statevector_order": "msq_first", "noisy_simulation": True}
-backend_info["qulacs"] = {"statevector_available": True, "statevector_order": "msq_first", "noisy_simulation": True}
-backend_info["cirq"] = {"statevector_available": True, "statevector_order": "lsq_first", "noisy_simulation": True}
-backend_info["qdk"] = {"statevector_available": False, "statevector_order": None, "noisy_simulation": False}
+class Simulator(SimulatorBase):
 
-
-class Simulator:
-
-    def __init__(self, target=default_simulator, n_shots=None, noise_model=None):
-        """Instantiate Simulator object.
+    def __new__(self, target=default_simulator, n_shots=None, noise_model=None, **kwargs):
```

Comment:
Take those wonderful type checks from `__init__` and use them here if you think they're useful.
Leave it alone and delete the typing imports if you think otherwise (NB: you were not consistent in your type checks in your own code file).